resolver.go

```diff
@@ -141,43 +142,44 @@ func (r *Resolver) iterateParents(ctx context.Context, qname, qtype string, dept
 		}
 
 		// Query all nameservers in parallel
-		count := 0
-		for _, nrr := range nrrs {
-			if nrr.Type != "NS" {
-				continue
-			}
-
-			go func(host string) {
-				rrs, err := r.exchange(ctx, host, qname, qtype, depth)
-				if err != nil {
-					chanErrs <- err
-				} else {
-					chanRRs <- rrs
+		if len(nrrs) > 0 {
+			count := 0
+			offset := rand.Intn(len(nrrs))
```

Comment:
The default random source in `math/rand` uses a global lock. Could we put the servers to query in a map and range over them for randomness instead?